tests/infra/remote.py

```diff
@@ -791,7 +791,10 @@ def __init__(
             exe_files += [config_file]
 
             with open(config_file, "w", encoding="utf-8") as f:
-                f.write(output)
+                # Parse and re-emit output to produce consistently formatted (indented) JSON.
+                # This will also ensure the render produced valid JSON
+                j = json.loads(output)
+                json.dump(j, f, indent=2)
```

Comment:
This is a kind-of-unrelated driveby change, but I think a useful QoL improvement. The `jinja`-rendered configs have some quirky formatting due to constraints in the original template, which make them harder to read by hand. This is a wastefully full re-encoding, but negligible in the startup time, and produces consistent formatting.

```
    "rpc_interfaces": {"primary_rpc_interface": {"app_protocol": "HTTP1", "bind_address": "127.0.0.1:8000", "endorsement": {"authority": "Service"}, "http_configuration": {"max_body_size": "1048576", "max_header_size": "16384", "max_headers_count": 256}, "max_open_sessions_hard": 1010, "max_open_sessions_soft": 1000, "protocol": "tcp", "published_address": "127.0.0.1:8000"}}
...
      "members": [{"certificate_file": "member0_cert.pem", "data_json_file": null, "encryption_public_key_file": "member0_enc_pubk.pem"}],
```
->
```
    "rpc_interfaces": {
      "primary_rpc_interface": {
        "app_protocol": "HTTP1",
        "bind_address": "127.0.0.1:8000",
        "endorsement": {
          "authority": "Service"
        },
        "forwarding_timeout_ms": 3000,
        "http_configuration": {
          "max_body_size": "1",
          "max_header_size": "2",
          "max_headers_count": 3
        },
        "max_open_sessions_hard": 1010,
        "max_open_sessions_soft": 1000,
        "protocol": "tcp",
        "published_address": "127.0.0.1:8000"
      }
    }
...
      "members": [
        {
          "certificate_file": "member0_cert.pem",
          "data_json_file": null,
          "encryption_public_key_file": "member0_enc_pubk.pem"
        }
      ],
```

Or more usefully
```
$ grep "max_header_size" workspace/sandbox_0/0.config.json
    "rpc_interfaces": {"primary_rpc_interface": {"app_protocol": "HTTP1", "bind_address": "127.0.0.1:8000", "endorsement": {"authority": "Service"}, "http_configuration": {"max_body_size": "1048576", "max_header_size": "16384", "max_headers_count": 256}, "max_open_sessions_hard": 1010, "max_open_sessions_soft": 1000, "protocol": "tcp", "published_address": "127.0.0.1:8000"}}
```
->
```
$ grep "max_header_size" workspace/sandbox_0/0.config.json
          "max_header_size": "2",
```